WordPress/wporg-main-2022
# Add download modal

See #40

This PR adds a modal window which is displayed when the 'Download WordPress' button is clicked on the download page. The download is also triggered as it is currently.

NOTE: this component depends on the list of links style which hasn't landed yet.

### Screenshots

![Screen Shot 2022-08-05 at 3 15 47 PM](https://user-images.githubusercontent.com/1017872/182993991-c35d1316-b672-4d1e-b685-cc2f83865fc6.jpg)

### How to test the changes in this Pull Request:

1. Ensure the download button has the id `wporg__download-button`.
2. Click the Download button and the modal should be displayed. The download should also start.
3. The links in the modal should open in the same window.
4. Close the modal by clicking the close button or overlay.
5. With your keyboard tab to the button and activate with enter. The behaviour should match clicking. The modal should receive focus so that it is easy to tab to the close button or the links. These should also be able to be activated with the keyboard. Focus should cycle within the modal while tabbing. 
6. Resize the screen to check the responsiveness.
